src/components/visualizations/common/utils/drilldownEventing.ts

```diff
@@ -0,0 +1,205 @@
+import { IDrillEventIntersectionElement } from '../../../../interfaces/DrillEvents';
+import { VisElementType, VisType, VisualizationTypes } from '../../../../constants/visualizationTypes';
+
+export interface IDrillableItemLocalId extends IDrillableItem {
+    localIdentifier: AFM.Identifier;
+}
+
+export function isDrillableItemLocalId(item: IDrillableItem | IDrillableItemLocalId): item is IDrillableItemLocalId {
+    return (item as IDrillableItemLocalId).localIdentifier !== undefined;
+}
+
+export interface IDrillIntersection {
+    id: string;
+    title: string;
+    value: Execution.DataValue;
+    name: string;
+    uri: string;
+    identifier: AFM.Identifier;
+}
+
+export interface IDrillPoint {
+    x: number;
+    y: number;
+    drillContext: IDrillIntersection[];
+}
+
+export interface IHCDrillEventGroup {
+    points: IDrillPoint[];
+}
+export interface IHCDrillEventPoint {
+    point: IDrillPoint;
+}
+
+export interface ICellDrillEvent {
+    columnIndex: number;
+    rowIndex: number;
+    row: string[];
+    intersection: IDrillIntersection[];
+}
+
+export type FiredDrillEventCallback = (data: any) => boolean;
+
+export interface IDrillConfig {
+    afm: AFM.IAfm;
+    onFiredDrillEvent: FiredDrillEventCallback;
+}
+
+export function isGroupHCDrillEvent(event: IHCDrillEventGroup | IHCDrillEventPoint): event is IHCDrillEventGroup {
+    return (event as IHCDrillEventGroup).points !== undefined;
+}
+
+function getPoPMeasureIdentifier(measure: AFM.IMeasure): AFM.Identifier {
+    return get<string>(measure, ['definition', 'popMeasure', 'measureIdentifier']);
+}
+
+function findMeasureByIdentifier(afm: AFM.IAfm, localIdentifier: AFM.Identifier) {
+    return (afm.measures || []).find((m: AFM.IMeasure) => m.localIdentifier === localIdentifier);
+}
+
+export function getMeasureUriOrIdentifier(afm: AFM.IAfm, localIdentifier: AFM.Identifier): IDrillableItem {
+    let measure = findMeasureByIdentifier(afm, localIdentifier);
+    if (measure) {
+        const popMeasureIdentifier = getPoPMeasureIdentifier(measure);
+        if (popMeasureIdentifier) {
+            measure = findMeasureByIdentifier(afm, popMeasureIdentifier);
+        }
+        return {
+            uri: get<string>(measure, ['definition', 'measure', 'item', 'uri']),
+            identifier: get<string>(measure, ['definition', 'measure', 'item', 'identifier'])
+        };
+    }
+    return null;
+}
+
+function isHeaderDrillable(drillableItems: IDrillableItem[], header: IDrillableItem) {
+    return drillableItems.some((drillableItem: IDrillableItem) =>
+        // Check for defined values because undefined === undefined
+        (drillableItem.identifier && header.identifier && drillableItem.identifier === header.identifier) ||
+        (drillableItem.uri && header.uri && drillableItem.uri === header.uri)
+    );
+}
+
+export function isDrillable(drillableItems: IDrillableItem[],
+                            header: IDrillableItemLocalId | IDrillableItem,
```

Comment:
use only `IDrillableItemLocalId` as it extends from `IDrillableItem`

Reply:
Impossible to simplify it to `IDrillableItemLocalId`